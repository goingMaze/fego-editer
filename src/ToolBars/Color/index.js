import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { EditorState, RichUtils, Modifier } from 'draft-js'
import { ToolBarBtn, Func, Picker } from 'utils'
let { toggleInlineStyle } = Func;

export default class Color extends Component {
	static propTypes = {
		editorState: PropTypes.object.isRequired,
		onChange: PropTypes.func.isRequired,
		config: PropTypes.array.isRequired,
		modalManage: PropTypes.object.isRequired
	}
	constructor(props) {
		super(props)
		this.state = {
			expanded: false
		}
	}
	componentWillMount() {
		let { modalManage } = this.props;
		modalManage.addCallback(this.changeExpand)
	}
	componentWillUnmount() {
		let { modalManage } = this.props;
		modalManage.removeCallback(this.changeExpand)
	}
	changeExpand = () => {
		this.setState({
			expanded: this.expanded
		})
		this.expanded = false
	}
	onExpand = () => {
		this.expanded = !this.state.expanded
	}
	onToggle = inlineStyle => {
		let { editorState, onChange, config } = this.props;
		config = config.map(item => `color${item}`)
		let newEditorState = toggleInlineStyle(editorState, config, `color${inlineStyle}`)
		onChange(newEditorState)
	}
	render() {
		const { editorState, config } = this.props;
		const { expanded } = this.state;
		const currentStyle = editorState.getCurrentInlineStyle();
		let activeItem = -1;
		config.find((style, idx) => {
			if (currentStyle.has(`color${style}`)) {
				activeItem = idx
				return true
			}
			return false
		})
		return (
			<div className="FegoEditor-toolbars" title='颜色' >
				{
					config.isTile ? config.map((style, idx) =>
						<ToolBarBtn
							key={idx}
							active={currentStyle.has(`color${style}`)}
							onToggle={this.onToggle}
							style={`color${style}`}
						>
							<svg className="icon" viewBox="0 0 1024 1024" style={{ color: style }} >
								<path d="M127.64823801 128.11282l766.89124798 0 0 766.891248-766.891248 0 0-766.891248Z" />
							</svg>
						</ToolBarBtn>
					) : 
					<Picker {...{ expanded, activeItem }} options={config} onToggle={this.onToggle} changeExpand={this.onExpand} style={{ maxHeight: '120px', width: '148px', overflow: 'auto' }} />
				}
			</div>
		)
	}
}