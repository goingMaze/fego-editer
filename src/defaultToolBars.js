const defaultToolBars = {
	options: ['BlockType', 'Color', 'FontFamily', 'FontSize', 'Image', 'Inline', 'Link', 'List', 'TextAlign', 'Video', 'History'],
	BgColor: [],
	BlockType: ['正文', 'H1', 'H2', 'H3', 'H4', 'H5', 'H6'],
	Color: ['rgb(97,189,109)', 'rgb(26,188,156)', 'rgb(84,172,210)', 'rgb(44,130,201)',
		'rgb(147,101,184)', 'rgb(71,85,119)', 'rgb(204,204,204)', 'rgb(65,168,95)', 'rgb(0,168,133)',
		'rgb(61,142,185)', 'rgb(41,105,176)', 'rgb(85,57,130)', 'rgb(40,50,78)', 'rgb(0,0,0)',
		'rgb(247,218,100)', 'rgb(251,160,38)', 'rgb(235,107,86)', 'rgb(226,80,65)', 'rgb(163,143,132)',
		'rgb(239,239,239)', 'rgb(255,255,255)', 'rgb(250,197,28)', 'rgb(243,121,52)', 'rgb(184,49,47)',
		'rgb(209,72,65)', 'rgb(124,112,107)', 'rgb(209,213,216)'],
	FontFamily: [{
		name: 'SimHei', family: '"SimHei", "黑体"'
	}, {
		name: 'Yahei', family: '"Microsoft Yahei", "微软雅黑"'
	}, {
		name: 'KaiTi', family: '"KaiTi", "楷体"'
	}, {
		name: 'FangSong', family: '"FangSong", "仿宋"'
	}],
	FontSize: [16, 20, 30, 40, 50, 70, 100, 200],
	History: ['undo', 'redo'],
	Image: {
		uploadUrl: '',
		uploadCallback: null
	},
	Inline: ['bold', 'italic', 'underline'],
	Link: {},
	List: ['ul', 'ol', 'indent', 'outdent'],
	TextAlign: ['left', 'center', 'right', 'justify'],
	Video: {},
	Remove: {}
};

export default defaultToolBars
