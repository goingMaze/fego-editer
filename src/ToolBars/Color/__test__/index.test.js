import React from 'react'
import { mount, render } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { EditorState, ContentState, convertFromHTML } from 'draft-js'
import Color from '..'
import defaultToolBars from '../../../defaultToolBars'
import ModalManage from '../../../utils/ModalManage'

describe('Color', () => {
	const contentState = ContentState.createFromBlockArray(convertFromHTML('<div>test</div>'))
	const editorState = EditorState.createWithContent(contentState)
	const wrapper = mount(
		<Color editorState={editorState} onChange={() => { }}
			config={defaultToolBars.Color} modalManage={new ModalManage()}
		/>
	);
	it('渲染时开头应该有一个div', () => {
		expect(wrapper.html().startsWith('<div')).toEqual(true)
	})
	it('应该有一个Picker子组件', () => {
		expect(wrapper.find('Picker').length).toBe(1)
	})
	it('config应该为默认值', () => {
		expect(wrapper.prop('config')).toEqual(defaultToolBars.Color)
	})
	it('在expanded为true时应该有28个icon元素', () => {
		expect(wrapper.find('.icon').length).toBe(1)
		wrapper.setState({
			expanded: true
		})
		expect(wrapper.find('.icon').length).toBe(28)
	})
	it('快照', () => {
		const rendered = render(
			<Color editorState={editorState} onChange={() => { }}
				config={defaultToolBars.Color} modalManage={new ModalManage()}
			/>
		)
		expect(toJSON(rendered)).toMatchSnapshot()
	})
})