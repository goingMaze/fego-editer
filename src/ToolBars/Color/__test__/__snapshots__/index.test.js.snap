// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color 快照 1`] = `
<div
  class="FegoEditor-toolbars"
  title="颜色"
>
  <div
    class="FegoEditor-picker"
  >
    <div
      class="FegoEditor-pickerLabel"
    >
      <svg
        aria-hidden="true"
        class="icon"
        style="color:#6a6f7b;"
        viewBox="0 0 1024 1024"
      >
        <path
          d="M470.975 137l-214.462 562.5h102l44.513-131.25h217.949l48.038 131.25h102.002l-214.501-562.5h-85.5zM512 256.55l78.525 233.175h-160.538l82.013-233.212zM212 774.5v112.5h599.999v-112.5h-599.999z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Color 快照 1`] = `
<div
  class="FegoEditor-toolbars"
  title="颜色"
>
  <div
    class="FegoEditor-picker"
  >
    <div
      class="FegoEditor-pickerLabel"
    >
      <svg
        aria-hidden="true"
        class="icon"
        style="color:#6a6f7b;"
        viewBox="0 0 1024 1024"
      >
        <path
          d="M470.975 137l-214.462 562.5h102l44.513-131.25h217.949l48.038 131.25h102.002l-214.501-562.5h-85.5zM512 256.55l78.525 233.175h-160.538l82.013-233.212zM212 774.5v112.5h599.999v-112.5h-599.999z"
        />
      </svg>
    </div>
  </div>
</div>
`;
